library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -47,103 +48,103 @@ public void migrate() {
-            Batch batch = migration.batch();
-
-            downloadsPersistence.startTransaction();
-
-            DownloadBatchTitle downloadBatchTitle = new LiteDownloadBatchTitle(batch.getTitle());
-            DownloadsBatchPersisted persistedBatch = new LiteDownloadsBatchPersisted(downloadBatchTitle, batch.getDownloadBatchId(), Status.DOWNLOADED);
-            downloadsPersistence.persistBatch(persistedBatch);
-
-            for (Migration.FileMetadata fileMetadata : migration.getFileMetadata()) {
-                String url = fileMetadata.uri();
-
-                FileName fileName = LiteFileName.from(batch, url);
-                FilePath filePath = FilePathCreator.create(fileName.name());
-                DownloadFileId downloadFileId = DownloadFileId.from(batch);
-                DownloadsFilePersisted persistedFile = new LiteDownloadsFilePersisted(
-                        batch.getDownloadBatchId(),
-                        downloadFileId,
-                        fileName,
-                        filePath,
-                        fileMetadata.fileSize().totalSize(),
-                        url,
-                        FilePersistenceType.INTERNAL
-                );
-                downloadsPersistence.persistFile(persistedFile);
-            }
-
-            downloadsPersistence.transactionSuccess();
-            downloadsPersistence.endTransaction();
+    private void migrateV1DataToV2Database(Migration migration) {
+        Batch batch = migration.batch();
+
+        DownloadBatchTitle downloadBatchTitle = new LiteDownloadBatchTitle(batch.getTitle());
+        DownloadsBatchPersisted persistedBatch = new LiteDownloadsBatchPersisted(downloadBatchTitle, batch.getDownloadBatchId(), Status.DOWNLOADED);
+        downloadsPersistence.persistBatch(persistedBatch);
+
+        for (Migration.FileMetadata fileMetadata : migration.getFileMetadata()) {
+            String url = fileMetadata.uri();
+
+            FileName fileName = LiteFileName.from(batch, url);
+            FilePath filePath = FilePathCreator.create(fileName.name());
+            DownloadFileId downloadFileId = DownloadFileId.from(batch);
+            DownloadsFilePersisted persistedFile = new LiteDownloadsFilePersisted(
+                    batch.getDownloadBatchId(),
+                    downloadFileId,
+                    fileName,
+                    filePath,
+                    fileMetadata.fileSize().totalSize(),
+                    url,
+                    FilePersistenceType.INTERNAL
+            );
+            downloadsPersistence.persistFile(persistedFile);
         }
     }
 
     // TODO: See https://github.com/novoda/download-manager/issues/270
-    private void deleteFrom(SqlDatabaseWrapper database, List<Migration> migrations) {
-        for (Migration migration : migrations) {
-            Batch batch = migration.batch();
-            database.rawQuery(DELETE_BY_ID_QUERY, batch.getDownloadBatchId().stringValue());
```

Comment:
This wasn't working before but it didn't matter because we just dropped the whole db anyway 😂